django/db/backends/base/schema.py

```diff
@@ -1660,7 +1660,9 @@ def _field_indexes_sql(self, model, field):
         return output
 
     def _field_should_be_altered(self, old_field, new_field, ignore=None):
-        if not old_field.concrete and not new_field.concrete:
+        if (not (old_field.concrete or old_field.many_to_many)) and (
+            not (new_field.concrete or new_field.many_to_many)
+        ):
```

Comment:
Makes sense to me. The migration framework supports alteration concrete field and `ManyToManyField` (which results in table alterations).

Reply:
I unfortunately I don't remember but I think your conclusion is sensible.